rust/connlib/tunnel/src/lib.rs

```diff
@@ -269,6 +285,14 @@ impl GatewayTunnel {
 
                     continue;
                 }
+                Poll::Ready(io::Input::UdpDnsQuery(query)) => self.io.send_udp_dns_response(
+                    query.source,
+                    dns::servfail(query.message.for_slice_ref()),
+                )?,
+                Poll::Ready(io::Input::TcpDnsQuery(query)) => self.io.send_tcp_dns_response(
+                    query.source,
+                    dns::servfail(query.message.for_slice_ref()),
+                )?,
```

Comment:
With this PR, the DNS server will be active but it always responds with SERVFAIL.